contracts/deploy/00_generate_configuration.ts

```diff
@@ -0,0 +1,40 @@
+import { generateGenesisFile } from "../scripts/create_genesis.ts";
+import { exec } from "child_process";
+import util from "node:util";
+import path from "path";
+import fs from "fs";
+
+const CLIENT_SBIN_DIR = `${__dirname}/../../clients/geth/specular/sbin`;
+const CLIENT_DATA_DIR = `${__dirname}/../../clients/geth/specular/data`;
+const execPromise = util.promisify(exec);
+
+const func: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
+  // 1. generate genesis file
+  const dataPath = `${__dirname}/../../clients/geth/specular/data/`;
+  const baseGenesisPath = path.join(dataPath, "base_genesis.json");
+  const genesisPath = path.join(dataPath, "genesis.json");
+
+  // 2. init geth nodes
+  await execPromise(path.join(CLIENT_SBIN_DIR, "clean.sh"));
+  await execPromise(path.join(CLIENT_SBIN_DIR, "init.sh"));
```

Comment:
@LEAFERx I saw that you added node initialisation to `export_genesis.sh` in your refactor branch. I think we can go either way but to me it seems cleaner to keep them separate. lmk what you think

Reply:
it is just your last pr can't work without init so I temporarily fix it by calling`init.sh` to `export_genesis.sh`. In fact we shouldn't call these `init.sh` etc because they're working on the dev sequencer data folder. we should use another temp data folder at least.

Reply:
I changed the export script. it creates a temporary data dir and deletes it upon completion